# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/gfx-renderdoc/src/lib.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/gfx-renderdoc/src/lib.rs b/third_party/rust/wgpu-hal/src/auxil/renderdoc.rs
rename from third_party/rust/gfx-renderdoc/src/lib.rs
rename to third_party/rust/wgpu-hal/src/auxil/renderdoc.rs
--- a/third_party/rust/gfx-renderdoc/src/lib.rs
+++ b/third_party/rust/wgpu-hal/src/auxil/renderdoc.rs
@@ -1,28 +1,21 @@
-#![warn(
-    trivial_casts,
-    trivial_numeric_casts,
-    unused_extern_crates,
-    unused_import_braces,
-    unused_qualifications
-)]
+//! RenderDoc integration - <https://renderdoc.org/>
 
-//! RenderDoc integration - https://renderdoc.org/
+use std::{ffi, os, ptr};
 
 /// The dynamically loaded RenderDoc API function table
 #[repr(C)]
 #[derive(Debug)]
 pub struct RenderDocApi {
     api: renderdoc_sys::RENDERDOC_API_1_4_1,
     lib: libloading::Library,
 }
 
 unsafe impl Send for RenderDocApi {}
-
 unsafe impl Sync for RenderDocApi {}
 
 /// RenderDoc API type
 #[derive(Debug)]
 pub enum RenderDoc {
     /// RenderDoc functionality is available
     Available {
         /// RenderDoc API with function pointers
@@ -32,17 +25,17 @@ pub enum RenderDoc {
     NotAvailable {
         /// A description why renderdoc functionality is not available
         reason: String,
     },
 }
 
 impl RenderDoc {
     pub unsafe fn new() -> Self {
-        type GetApiFn = unsafe extern "C" fn(version: u32, out: *mut *mut std::ffi::c_void) -> i32;
+        type GetApiFn = unsafe extern "C" fn(version: u32, out: *mut *mut ffi::c_void) -> i32;
 
         #[cfg(windows)]
         let renderdoc_filename = "renderdoc.dll";
         #[cfg(all(unix, not(target_os = "android")))]
         let renderdoc_filename = "librenderdoc.so";
         #[cfg(target_os = "android")]
         let renderdoc_filename = "libVkLayer_GLES_RenderDoc.so";
 
@@ -64,17 +57,17 @@ impl RenderDoc {
                 return RenderDoc::NotAvailable {
                     reason: format!(
                         "Unable to get RENDERDOC_GetAPI from renderdoc library '{}': {:?}",
                         renderdoc_filename, e
                     ),
                 }
             }
         };
-        let mut obj = std::ptr::null_mut();
+        let mut obj = ptr::null_mut();
         match get_api(10401, &mut obj) {
             1 => RenderDoc::Available {
                 api: RenderDocApi {
                     api: *(obj as *mut renderdoc_sys::RENDERDOC_API_1_4_1),
                     lib: renderdoc_lib,
                 },
             },
             return_value => RenderDoc::NotAvailable {
@@ -93,34 +86,36 @@ impl Default for RenderDoc {
             return RenderDoc::NotAvailable {
                 reason: "RenderDoc support is only enabled with 'debug_assertions'".into(),
             };
         }
         unsafe { Self::new() }
     }
 }
 /// A implementation specific handle
-pub type Handle = *mut ::std::os::raw::c_void;
+pub type Handle = *mut os::raw::c_void;
 
 impl RenderDoc {
     /// Start a RenderDoc frame capture
-    pub unsafe fn start_frame_capture(&self, device_handle: Handle, window_handle: Handle) {
-        match self {
+    pub unsafe fn start_frame_capture(&self, device_handle: Handle, window_handle: Handle) -> bool {
+        match *self {
             Self::Available { api: ref entry } => {
                 entry.api.StartFrameCapture.unwrap()(device_handle, window_handle);
+                true
             }
             Self::NotAvailable { ref reason } => {
-                log::warn!("Could not start RenderDoc frame capture: {}", reason)
+                log::warn!("Could not start RenderDoc frame capture: {}", reason);
+                false
             }
-        };
+        }
     }
 
     /// End a RenderDoc frame capture
     pub unsafe fn end_frame_capture(&self, device_handle: Handle, window_handle: Handle) {
-        match self {
+        match *self {
             Self::Available { api: ref entry } => {
                 entry.api.EndFrameCapture.unwrap()(device_handle, window_handle);
             }
             Self::NotAvailable { ref reason } => {
                 log::warn!("Could not end RenderDoc frame capture: {}", reason)
             }
         };
     }